src/Utility/worker.js

```diff
@@ -83,41 +84,42 @@ function objEquiv(a, b, opts) {
   return typeof a === typeof b;
 }
 
-class Test {
-    constructor(params) {
-        this.tests = params.tests;
-        this.solution = params.solution;
-        this.results = [];
-        this.passed = true;
-
-        this.tests.forEach(({descriptor, argumentsToPass, expectedResult}) => {
-
-            argumentsToPass = argumentsToPass.map(arg => {
-                if(typeof arg === 'array' || typeof arg === 'object' || typeof arg === 'string') {
-                    return JSON.stringify(arg);
-                } else {
-                    return arg;
-                }
-            });
-
-            
-            const solution = eval(`(${this.solution}\n)`);
-            const result = eval(`(solution(${argumentsToPass}))`);
-            if(!deepEqual(result, expectedResult)) this.passed = false;
-
-            this.results.push({
-                descriptor,
-                passed: deepEqual(result, expectedResult)
-            });
-        });
-    }
-}
 
+////////////////////////////////////////////////////////////////////////
+function Testing(params){
+  const tests = params.tests;
+  const userSolution = params.solution;
+  const results = [];
+ 
+  tests.forEach(({descriptor, argumentsToPass, expectedResult}) => {
+
+    argumentsToPass = argumentsToPass.map(arg => {
+        if(typeof arg === 'array' || typeof arg === 'object' || typeof arg === 'string') {
```

Comment:
I realize it's my bad, but I realized after writing `typeof arg === 'array'` that `typeof [] === 'object'` .. one of those Javascript gotchas. You can remove this part of the if statement.